packages/nouns-contracts/contracts/governance/NounsDAOLogicV2.sol

```diff
@@ -0,0 +1,895 @@
+        );
+        require(
+            newMinQuorumVotesBPS <= params.maxQuorumVotesBPS,
+            'NounsDAO::_setMinQuorumVotesBPS: min quorum votes bps greater than max'
+        );
+
+        uint16 oldMinQuorumVotesBPS = params.minQuorumVotesBPS;
+        params.minQuorumVotesBPS = newMinQuorumVotesBPS;
+
+        _writeQuorumParamsCheckpoint(params);
+
+        emit MinQuorumVotesBPSSet(oldMinQuorumVotesBPS, newMinQuorumVotesBPS);
+    }
+
+    function _setMaxQuorumVotesBPS(uint16 newMaxQuorumVotesBPS) external {
+        require(msg.sender == admin, 'NounsDAO::_setMaxQuorumVotesBPS: admin only');
+        DynamicQuorumParams memory params = getDynamicQuorumParamsAt(block.number);
+
+        require(
+            newMaxQuorumVotesBPS <= MAX_QUORUM_VOTES_BPS_UPPER_BOUND,
+            'NounsDAO::_setMaxQuorumVotesBPS: invalid max quorum votes bps'
+        );
+        require(
+            params.minQuorumVotesBPS <= newMaxQuorumVotesBPS,
+            'NounsDAO::_setMaxQuorumVotesBPS: min quorum votes bps greater than max'
+        );
+
+        uint16 oldMaxQuorumVotesBPS = params.maxQuorumVotesBPS;
+        params.maxQuorumVotesBPS = newMaxQuorumVotesBPS;
+
+        _writeQuorumParamsCheckpoint(params);
+
+        emit MaxQuorumVotesBPSSet(oldMaxQuorumVotesBPS, newMaxQuorumVotesBPS);
+    }
+
+    function _setQuorumVotesBPSOffset(uint16 newQuorumVotesBPSOffset) external {
+        require(msg.sender == admin, 'NounsDAO::_setQuorumVotesBPSOffset: admin only');
+        DynamicQuorumParams memory params = getDynamicQuorumParamsAt(block.number);
+
+        uint16 oldQuorumVotesBPSOffset = params.quorumVotesBPSOffset;
+        params.quorumVotesBPSOffset = newQuorumVotesBPSOffset;
+
+        _writeQuorumParamsCheckpoint(params);
+
+        emit QuorumVotesBPSOffsetSet(oldQuorumVotesBPSOffset, newQuorumVotesBPSOffset);
+    }
+
+    function _setQuorumLinearCoef(uint32 newQuorumLinearCoef) external {
+        require(msg.sender == admin, 'NounsDAO::_setQuorumLinearCoef: admin only');
+        DynamicQuorumParams memory params = getDynamicQuorumParamsAt(block.number);
+
+        uint32 oldQuorumLinearCoef = params.quorumLinearCoef;
+        params.quorumLinearCoef = newQuorumLinearCoef;
+
+        _writeQuorumParamsCheckpoint(params);
+
+        emit QuorumLinearCoefSet(oldQuorumLinearCoef, newQuorumLinearCoef);
+    }
+
+    function _setQuorumQuadraticCoef(uint32 newQuorumQuadraticCoef) external {
+        require(msg.sender == admin, 'NounsDAO::_setQuorumQuadraticCoef: admin only');
+        DynamicQuorumParams memory params = getDynamicQuorumParamsAt(block.number);
+
+        uint32 oldQuorumQuadraticCoef = params.quorumQuadraticCoef;
+        params.quorumQuadraticCoef = newQuorumQuadraticCoef;
+
+        _writeQuorumParamsCheckpoint(params);
+
+        emit QuorumQuadraticCoefSet(oldQuorumQuadraticCoef, newQuorumQuadraticCoef);
+    }
+
+    function _setDynamicQuorumParams(DynamicQuorumParams memory params) public {
```

Comment:
Should be able to add function comments to these now. Public interfaces look good.

Reply:
added: https://github.com/nounsDAO/nouns-monorepo/pull/438/commits/951d57faa1096fdc82df3d7261fdc4d98570e3f3